### ISSUE TITLE
Carousel doesn't start working till browser resize gets to a point where the carousel needs to resize, then works as intended.

### ISSUE THREAD

**Issue by DarceyLloyd** [davidcetinkaya/embla-carousel#195]
### Bug is related to

Embla wont init till resize is triggered.

- [ ] Vanilla 4.5.3
- [ ] ES6 Webpack Project

### Embla Carousel version

- v4.5.3

### Describe the bug

- Carousel doesn't become interactive till a browser resize to the carousel's width is triggered.

### Code
- HTML
```
    <div id="c1" class="embla embla-1">
        <div class="embla__viewport">
            <div class="embla__container">
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/01-profile.jpg" />
                    </div>
                </div>
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/02-profile.jpg" />
                    </div>
                </div>
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/03-profile.jpg" />
                    </div>
                </div>
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/04-profile.jpg" />
                    </div>
                </div>
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/05-profile.jpg" />
                    </div>
                </div>
                <div class="embla__slide">
                    <div class="embla__slide__inner">
                        <img class="embla__slide__img" src="/assets/images/artist-pages/06-profile.jpg" />
                    </div>
                </div>
            </div>
        </div>
    </div>
```
- JS
```
const mainCarouselWrap = document.getElementById("c1");
        const mainCarouselView = mainCarouselWrap.querySelector("c1 .embla__viewport");
        const mainCarousel = EmblaCarousel(mainCarouselView, {
            selectedClass: "",
            slidesToScroll: 2,
            loop: false
        });
```
- CSS
```
.embla {
            position: relative;
            background-color: #f7f7f7;
            padding: 20px;
            max-width: 670px;
            margin-left: auto;
            margin-right: auto;
        }

        .embla__viewport {
            overflow: hidden;
            width: 100%;
        }

        .embla__viewport.is-draggable {
            cursor: move;
            cursor: grab;
        }

        .embla__viewport.is-dragging {
            cursor: grabbing;
        }

        .embla__container {
            display: flex;
            user-select: none;
            -webkit-touch-callout: none;
            -khtml-user-select: none;
            -webkit-tap-highlight-color: transparent;
            margin-left: -10px;
        }

        .embla__slide {
            position: relative;
            min-width: 50%;
            padding-left: 10px;
        }

        .embla__slide__inner {
            position: relative;
            overflow: hidden;
            height: 190px;
        }

        .embla__slide__img {
            position: absolute;
            display: block;
            top: 50%;
            left: 50%;
            width: auto;
            min-height: 100%;
            min-width: 100%;
            max-width: none;
            transform: translate(-50%, -50%);
        }
```

Video: 
https://youtu.be/Ff2r0oNi9hc

**Comment by davidcetinkaya** [comment]
Hi @DarceyLloyd,

Please provide a CodeSandbox with your setup that reproduces the issue. I need that in order to debug properly. You can fork one of the CodeSandbox on the [examples page](https://www.embla-carousel.com/examples/basic).

This is most likely happening because you’re initializing Embla before you’ve rendered the correct sizes of the carousel elements. For example, if you’re toggling `display: none` on some parent element or similar.

Kindly,
David

**Comment by DarceyLloyd** [comment]
Hi David,

Sorry can't create a sandbox of this one as it's inside a project under NDA, but the functionality is:
- base html page is loaded
- there is a container for overlay content where a html file is xmlhttprequest use to load the page and injected into the container.
- once the xhr request succeeds the html is then injected into the container
- then the corresponding javascript class is executed to init on the new html content, picking up the html elements etc and then running embla setup
- during this time however the content container is display none, opacity: 0 and then GSAP tweened in from display: block, opacity: 1

Is there a force init command for embla? I guess I could slap a setTimout on the embla init function to delay the setup till the display: block and opacity: starts to tween in to 1...

**Comment by DarceyLloyd** [comment]
I put a setTimout of 1000ms on the init and it works but it does do a little stutter.
https://youtu.be/3Ba5R2DzDlQ

**Comment by davidcetinkaya** [comment]
Hi @DarceyLloyd,

Thanks for the additional information. As I suspected. `display: none` makes it impossible for Embla to pick up the carousel container and slide dimensions, because these elements won’t have any dimensions until `display: none` is removed.

My recommendation is as you’ve already figured out, to only animate opacity. You can use the [`reInit`](https://www.embla-carousel.com/api/options/#reinit) to re-initialize Embla and tell it to re-calculate all dimensions.

Best,
David

**Comment by davidcetinkaya** [comment]
@DarceyLloyd let me know how it goes. Cheers.

**Comment by DarceyLloyd** [comment]
@davidcetinkaya all good now thanks, the reInit did help prevent the setup glitch, but after applying the opacity fade, it adds a nice effect to the page to see the carousels fade (0.3s via GSAP )  👍 

Thanks again for the info :)

Embla is my go to carousel, I've never been a fan of the others, and I wont touch anything jQuery. Embla is the closes way to how I would build a carousel and expect it to be init and behave. Also a 1/4 of the css the others take.

Keep up the great work.

Thanks
Darcey

**Comment by dalalRohit** [comment]
How did you solved this?
I'm also facing the same error ;/

**Comment by davidcetinkaya** [comment]
@dalalRohit the following issues might help you:

- #159 
- #210

**Comment by davidjerleke** [comment]
This will be handled automatically in the next release:

- #159

**Comment by davidjerleke** [comment]
@DarceyLloyd, @dalalRohit,

This feature has been released with [v8.0.0-rc01](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc01).
